pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -48,15 +52,60 @@ def __init__(self):
         # method
         self.latent_to_symptom_delay =\
             pe.Parameters.instance().latent_to_sympt_delay
+        # Defining the length of the model time step (in days, can be a
+        # fraction of day as well).
         self.model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
         self.delay = np.floor(self.latent_to_symptom_delay
                               / self.model_time_step)
 
+        # Infectiousness progression
+        # Instantiate parameters to be used in update infectiousness
+        infectious_profile = pe.Parameters.instance().infectiousness_prof
+        inf_prof_resolution = len(infectious_profile) - 1
+        infectious_period = pe.Parameters.instance().asympt_infect_period
+        # Extreme case where model time step would be too small
+        max_inf_steps = 2550
+        # Define number of time steps a person is infectious:
+        num_infectious_ts =\
+            int(np.ceil(infectious_period / self.model_time_step))
+        if num_infectious_ts >= max_inf_steps:
+            raise ValueError('Number of infect timesteps exceeds limit')
```

Comment:
I'm not sure what you mean here but I don't think its clear - can you rephrase the error message? Maybe "Number of timesteps in infectious period exceeds limit - please reduce `time_steps_per_day`"?